eo-runtime/src/test/java/org/eolang/PhDefaultTest.java

```diff
@@ -483,6 +483,101 @@ void failsCorrectlyWhenTooManyAttributesPut() {
         );
     }
 
+    @Test
+    void verifiesThreadLocalNesting() {
+        final Phi phi = new PhDefaultTest.Int();
+        Assertions.assertDoesNotThrow(
+            () -> phi.take("context-verifiesThreadLocalNesting"),
+            "Nesting should be properly managed without exceptions"
+        );
+    }
+
+    @Test
+    void verifiesThreadLocalNestingWithExceptions() {
+        final Phi phi = new PhDefaultTest.Int();
+        Assertions.assertThrows(
+            ExUnset.class,
+            () -> phi.take("non-existent-attribute"),
+            "Should throw exception for non-existent attribute"
+        );
+        Assertions.assertDoesNotThrow(
+            () -> phi.take("context-verifiesThreadLocalNestingWithExceptions"),
+            "Should still work after exception"
+        );
+    }
+
+    @Test
+    void verifiesCleanupNesting() {
```

Comment:
@karishka1222 maybe we can rename the test to `worksAfterCleanup`? With the current name, it looks like we testing our `cleanUpNesting()` private method, specified in tests. WDYT?